packages/modules/pricing/src/services/pricing-module.ts

```diff
@@ -182,21 +181,20 @@ export default class PricingModuleService
       return [priceSets, count]
     }
 
-    const priceSetIds: string[] = []
-    const priceSetMap = new Map()
-    for (const priceSet of priceSets) {
-      priceSetIds.push(priceSet.id)
-      priceSetMap.set(priceSet.id, priceSet)
-    }
-
     const calculatedPrices = await this.calculatePrices(
-      { id: priceSetIds },
+      { id: priceSets.map((p) => p.id) },
       { context: pricingContext },
       sharedContext
     )
+
+    const calculatedPricesMap = new Map()
     for (const calculatedPrice of calculatedPrices) {
-      const priceSet = priceSetMap.get(calculatedPrice.id)
-      priceSet.calculated_price = calculatedPrice
+      calculatedPricesMap.set(calculatedPrice.id, calculatedPrice)
+    }
+
+    for (const priceSet of priceSets) {
```

Comment:
We swapped to iterating over the price set so we can set the calculated price explicitly to null (and not undefined).